pages/hub/insights/index.tsx

```diff
@@ -67,23 +115,38 @@ const InsightsHub: WithPageLayout = () => {
   }
 
   useEffect(() => {
-    if (session && session.insights_count === 0 && !dismissFeaturedInsights) {
+    if (
+      (session && session.insights_count === 0 && !dismissFeaturedInsights) ||
+      (!session && !dismissFeaturedInsights)
```

Comment:
Refactor (blocking): This covers the `!session && !dismissFeaturedInsights` case as well as the check for the insights count.

```suggestion
      session?.insights_count === 0 && !dismissFeaturedInsights
```

Reply:
The session can be type of `DbUser | false`, so if we use `session?.insights_count`, typescript screams with a type error  
<img width="513" alt="image" src="https://github.com/open-sauced/app/assets/62995161/6081b516-ffc0-4f06-91fa-af8883a2f1d2">

hence the reason i added the `!session && !dismissFeaturedInsights` check as well. let me know if there are ways to optimize in this case